src/Compilers/CSharp/Test/Emit2/Semantics/CollectionLiteralTests.cs

```diff
@@ -5644,6 +5667,2548 @@ private static void VerifyTypes(SemanticModel model, ExpressionSyntax expr, stri
             Assert.Equal(expectedConversionKind, conversion.Kind);
         }
 
+        [CombinatorialData]
+        [ConditionalTheory(typeof(CoreClrOnly))]
+        public void CollectionBuilder_01(bool useCompilationReference)
+        {
+            string sourceA = """
+                using System;
+                using System.Collections;
+                using System.Collections.Generic;
+                using System.Runtime.CompilerServices;
+                [CollectionBuilder(typeof(MyCollectionBuilder), "Create")]
+                public struct MyCollection<T> : IEnumerable<T>
+                {
+                    private readonly List<T> _list;
+                    public MyCollection(List<T> list) { _list = list; }
+                    public IEnumerator<T> GetEnumerator() => _list.GetEnumerator();
+                    IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+                }
+                public class MyCollectionBuilder
+                {
+                    public static MyCollection<T> Create<T>(ReadOnlySpan<T> items)
```

Comment:
ideas for tests (if not already here):

1. additional optional parameters after the ROS parameter.
2. a params parameter after it.
3. (in the future, if we allow `params ROS` we should ensure that works for these methods.
4. first arg is not ROS.
5. Return type is convertible (or identity-reference convertible) type to the original type.

For the last one, the case is:

```c#
[CollectionBuilder(typeof(ImmutableHashSet), "Create")]
public interface IImutableSet<T> { ... }

public static class ImmutableHashSet
{
    public static ImmutableHashSet<T> Create(ReadOnlySpan<T> items)
}
```

Here `ImmutableHashSet<T>` is ref-identity convertible to `IImutableSet<T>`

Reply:
test with Create method having constraints taht may or may not match the callsite type (like if it is constrained on `struct`).

Reply:
> ref-identity convertible

I have no idea what ref-identity convertible means. Did you just mean an implicit reference conversion?

Reply:
yes.